packages/cli/src/commands/dev.js

```diff
@@ -48,7 +49,11 @@ export default {
     await nuxt.ready()
 
     // Start listening
-    await nuxt.server.listen()
+    await nuxt.server.listen(this._usedPort)
```

Comment:
What if `server.port` in `nuxt.config` is changed at runtime ?

Reply:
The `server.port` may be in use when we used the random port. checking for that value means we again fall back trying a used port and then choosing another random one. 

Alternatively, we can preserve a random chosen PORT in a global variable inside `listener` module and reuse it.

(PS: I think this fix can be moved to another PR for better history)

Reply:
Will create another PR for this.

Reply:
Let's move to #5793 5793